lib/rules/no-restricted-v-on.js

```diff
@@ -0,0 +1,192 @@
+    return (s) => {
+      re.lastIndex = 0
+      return re.test(s)
+    }
+  }
+  return (s) => s === str
+}
+
+/**
+ * @param {any} option
+ * @returns {ParsedOption}
+ */
+function parseOption(option) {
+  if (typeof option === 'string') {
+    const matcher = buildMatcher(option)
+    return {
+      test(key) {
+        return Boolean(
+          key.argument &&
+            key.argument.type === 'VIdentifier' &&
+            matcher(key.argument.rawName)
+        )
+      },
+      modifiers: []
+    }
+  }
+  if (option === null) {
+    return {
+      test(key) {
+        return key.argument === null
+      },
+      modifiers: []
+    }
+  }
+  const parsed = parseOption(option.argument)
+
+  if (option.modifiers) {
+    const argTest = parsed.test
+    parsed.test = (key) => {
+      if (!argTest(key)) {
+        return false
+      }
+      return /** @type {string[]} */ (option.modifiers).every((modName) =>
+        key.modifiers.some((mid) => mid.name === modName)
+      )
+    }
+    parsed.modifiers = option.modifiers
+  }
+  if (option.element) {
+    const argTest = parsed.test
+    const tagMatcher = buildMatcher(option.element)
+    parsed.test = (key) => {
+      if (!argTest(key)) {
+        return false
+      }
+      return tagMatcher(key.parent.parent.parent.rawName)
+    }
+    parsed.useElement = true
+  }
+  parsed.message = option.message
+  return parsed
+}
+
+/**
+ * @param {VDirectiveKey} key
+ * @param {ParsedOption} option
+ */
+function defaultMessage(key, option) {
+  const von = key.name.rawName === '@' ? '' : 'v-on'
+  const arg =
+    key.argument != null && key.argument.type === 'VIdentifier'
+      ? `@${key.argument.rawName}`
+      : ''
+  const mod =
+    option.modifiers.length > 0 ? `.${option.modifiers.join('.')}` : ''
+  let element = 'element'
+  if (option.useElement) {
+    element = `<${key.parent.parent.parent.rawName}>`
+  }
+  return `Using \`${von + arg + mod}\` is not allowed on this ${element}.`
+}
+
+module.exports = {
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description: 'disallow specific argument in `v-on`',
+      categories: undefined,
+      url: 'https://eslint.vuejs.org/rules/no-restricted-v-on.html'
+    },
+    fixable: null,
+    schema: {
+      type: 'array',
+      items: {
+        oneOf: [
+          { type: ['string', 'null'] },
+          {
+            type: 'object',
+            properties: {
+              argument: { type: 'string' },
+              element: { type: 'string' },
+              message: { type: 'string', minLength: 1 },
+              modifiers: {
+                type: 'array',
+                items: {
+                  type: 'string',
+                  enum: [
+                    'prevent',
+                    'stop',
+                    'capture',
+                    'self',
+                    'once',
+                    'passive'
+                  ]
+                },
+                uniqueItems: true
```

Comment:
If the `modifiers` array is set, it should contain at least one item:

```suggestion
                uniqueItems: true,
                minItems: 1
```